test/unit/specs/vislib/lib/y_axis.js

```diff
@@ -210,128 +225,132 @@ define(function (require) {
-          expect(domain[1]).to.be(max);
+        it('should throw a NaN error', function () {
+          min = 'Not a number';
+          max = 12;
+
+          expect(function () {
+            yAxis._validateUserExtents(min, max);
+          }).to.throwError();
+        });
+
+        it('should return a decimal value', function () {
+          yAxis._attr.mode = 'percentage';
+          yAxis._attr.setYExtents = true;
+          domain = [];
+          domain[0] = yAxis._attr.yAxis.min = 20;
+          domain[1] = yAxis._attr.yAxis.max = 80;
+          newDomain = yAxis._validateUserExtents(domain);
+
+          expect(newDomain[0]).to.be(domain[0] / 100);
+          expect(newDomain[1]).to.be(domain[1] / 100);
+        });
+
+        it('should return the user defined value', function () {
+          domain = [20, 50];
+          newDomain = yAxis._validateUserExtents(domain);
+
+          expect(newDomain[0]).to.be(domain[0]);
+          expect(newDomain[1]).to.be(domain[1]);
+        });
+      });
+
+      describe('should throw an error when', function () {
+        var min;
+        var max;
+
+        it('min === max', function () {
+          min = 12;
+          max = 12;
+
+          expect(function () {
+            yAxis._validateAxisExtents(min, max);
+          }).to.throwError();
+        });
+
+        it('min > max', function () {
+          min = 30;
+          max = 10;
+
+          expect(function () {
+            yAxis._validateAxisExtents(min, max);
+          }).to.throwError();
         });
       });
     });
 
+    //describe('formatAxisLabel method', function () {
+    //  var num = 1e9;
+    //  var val;
+    //
+    //  describe('should not return a nice scale when defaultYExtents is true', function () {
+    //    beforeEach(function () {
+    //      createData(defaultGraphData);
+    //      yAxis._attr.defaultYExtents = true;
+    //      yAxis.getYAxis(height);
+    //      yAxis.render();
+    //    });
+    //
+    //    it('not return a nice scale', function () {
+    //      var min = _.min(_.flatten(defaultGraphData));
+    //      var max = _.max(_.flatten(defaultGraphData));
+    //      var domain = yAxis.yAxis.scale().domain();
+    //      expect(domain[0]).to.be(min);
+    //      expect(domain[1]).to.be(max);
+    //    });
+    //  });
+    //});
+
     describe('getScaleType method', function () {
       var fnNames = ['linear', 'log', 'square root'];
 
       it('should return a function', function () {
         fnNames.forEach(function (fnName) {
-          var isFunction = (typeof yAxis.getScaleType(fnName) === 'function');
+          var isFunction = (typeof yAxis._getScaleType(fnName) === 'function');
           expect(isFunction).to.be(true);
         });
 
         // if no value is provided to the function, scale should default to a linear scale
-        expect(typeof yAxis.getScaleType()).to.be('function');
+        expect(typeof yAxis._getScaleType()).to.be('function');
```

Comment:
Can you change this to `expect(yAxis._getScaleType()).to.be.a(Function);` here? Same on line 320